tools/downloader/topologies.py

```diff
@@ -0,0 +1,196 @@
+import sys
+import os
+import re
+import yaml
+import types
+import argparse
+
+from pathlib import Path
+
+import common
+
+_precision_marker = '$precision'
+_cache = Path(os.getenv('OPENCV_OPEN_MODEL_ZOO_CACHE_DIR', Path('.').resolve() / 'models'))
+
+
+class Topology:
+    def __init__(self, name, task_type, download_dir, config, model, framework,
+                 mo_args, precision):
+        self.name = name
+        self.task_type = task_type
+        self.mo_args = mo_args
+        self.download_dir = download_dir
+        self.config = config.replace(_precision_marker, precision)
+        self.model = model.replace(_precision_marker, precision)
+        self.framework = framework
+
+        if self.config and not os.path.exists(self.config) or \
+           self.model and not os.path.exists(self.model):
+            sys.argv = ['', '--name=' + self.name, '--precisions=' + precision,
+                        '--output_dir=' + str(_cache).replace('\\', '\\\\')]
+            from downloader import main
+            main()
```

Comment:
That's a good point. Thanks! Replaced to direct main invocation (with parser arguments as parameters).

Reply:
Great, this addresses 1 point out of 3. What about the rest of them?

Seriously, the downloader is not a library. It is a program. Use it as a program.

Reply:
I agree with @IRDonch on this one. The current solution feels hacky, and not robust. I believe that there is a need for a library that loads model zoo models, aside from having a __program__ that downloads models, but let me elaborate on that in a separate thread. It might be that we need a more significant refactor to achieve the two goals (downloader program & library). Let's figure that out once we have figured out what exactly we want out of the library - so let me get feedback on that one first :)